File: FLM_SubconLabelSystem/Library/Library.Database/BLL/LotSlitting.vb
Namespace BLL
    ''' <summary>
    ''' Business Logic Layer
    ''' ---------------------------------
    ''' 18 Feb 2012   Yeon    Initial Version
    ''' </summary>
    ''' <remarks></remarks>
    Public Class LotSlitting
        Inherits Library.Root.Other.BusinessLogicBase


        Public Shared Function List(ByVal Table As String, ByVal TableID As String, ByVal SearchField As String, ByVal SearchValue As String, ByVal SortField As String, ByVal Direction As Integer, _
                                    ByVal Page As Integer, ByVal Deleted As Integer) As ListCollection
            Using _dal As New DAL.LotSlitting
                'Validation the parameter value                
                If Direction <> 1 Then
                    Direction = 0
                End If
                List = _dal.List(Table, TableID, SearchField, SearchValue, SortField, Direction, FromRowNo(Page), ToRowNo(Page), Deleted)
            End Using
        End Function

        Public Shared Function GetData(ByVal ID As String) As DataTable
            Using _dal As New DAL.LotSlitting
                GetData = _dal.GetData(ID)
            End Using
        End Function

        'Public Shared Function GetDLLData(ByVal Value As String, ByVal ID As String) As DataTable
        '    Using _dal As New DAL.user
        '        GetDLLData = _dal.GetDLLData(Value, ID)
        '    End Using
        'End Function

        Public Shared Function Maint(ByVal ID As String, ByVal LOTNO As String, ByVal var2 As String, ByVal var3 As String, ByVal var4 As String, _
                                     ByVal var5 As String, ByVal var6 As String, ByVal var7 As String, _
                                     ByVal var8 As String, ByVal var9 As String, ByVal var10 As String, _
                                     ByVal var11 As String, ByVal var12 As String, ByVal RecType As String) As String
            Using _Dal As New DAL.LotSlitting
                Dim str As String = System.Web.HttpContext.Current.Session("gstrUserID").ToString
                'Dim cc As String = System.Web.HttpContext.Current.Session("gstrUserCompCode").ToString
                Maint = _Dal.Maint(ID, LOTNO, var2, var3, var4, var5, var6, var7, var8, var9, var10, var11, var12, RecType, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString)

                If Maint = "1" Then
                    _Dal.Commit()
                Else
                    _Dal.Rollback()
                End If
            End Using
        End Function

        Public Shared Function UpdPrintSel(ByVal SLITLOTNO As String, ByVal PrintSel As Boolean, _
                                           ByVal RecUpd As String) As String
            Using _Dal As New DAL.LotSlitting
                Dim str As String = System.Web.HttpContext.Current.Session("gstrUserID").ToString
                'Dim cc As String = System.Web.HttpContext.Current.Session("gstrUserCompCode").ToString
                UpdPrintSel = _Dal.UpdPrintSel(SLITLOTNO, PrintSel, RecUpd, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString)

                If UpdPrintSel = "1" Then
                    _Dal.Commit()
                Else
                    _Dal.Rollback()
                End If
            End Using
        End Function

        Public Shared Function UpdPrintSelAll(ByVal PrintSel As Boolean, _
                                           ByVal RecUpd As String, ByVal filter As String, ByVal filterfield As String, ByVal addCondition As String, ByVal passType As String) As String
            Using _Dal As New DAL.LotSlitting
                Dim str As String = System.Web.HttpContext.Current.Session("gstrUserID").ToString
                'Dim cc As String = System.Web.HttpContext.Current.Session("gstrUserCompCode").ToString
                UpdPrintSelAll = _Dal.UpdPrintSelAll(PrintSel, RecUpd, filter, filterfield, addCondition, passType, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString)

                If UpdPrintSelAll = "1" Then
                    _Dal.Commit()
                Else
                    _Dal.Rollback()
                End If
            End Using
        End Function
    End Class
End Namespace

File: FLM_SubconLabelSystem/Library/Library.Database/BLL/LotSlitting.cs
using System.Data;

namespace Library.Database.BLL
{
    /// <summary>
    /// Business Logic Layer
    /// ---------------------------------
    /// 18 Feb 2012   Yeon    Initial Version
    /// </summary>
    public class LotSlitting : Library.Root.Other.BusinessLogicBase
    {
        public static ListCollection List(string Table, string TableID, string SearchField, string SearchValue, string SortField, int Direction,
                                          int Page, int Deleted)
        {
            using (var _dal = new DAL.LotSlitting())
            {
                if (Direction != 1)
                {
                    Direction = 0;
                }
                return _dal.List(Table, TableID, SearchField, SearchValue, SortField, Direction, FromRowNo(Page), ToRowNo(Page), Deleted);
            }
        }

        public static DataTable GetData(string ID)
        {
            using (var _dal = new DAL.LotSlitting())
            {
                return _dal.GetData(ID);
            }
        }

        public static string Maint(string ID, string LOTNO, string var2, string var3, string var4,
                                    string var5, string var6, string var7,
                                    string var8, string var9, string var10,
                                    string var11, string var12, string RecType)
        {
            using (var _Dal = new DAL.LotSlitting())
            {
                string str = System.Web.HttpContext.Current.Session["gstrUserID"].ToString();
                string result = _Dal.Maint(ID, LOTNO, var2, var3, var4, var5, var6, var7, var8, var9, var10, var11, var12, RecType, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString());

                if (result == "1")
                {
                    _Dal.Commit();
                }
                else
                {
                    _Dal.Rollback();
                }
                return result;
            }
        }

        public static string UpdPrintSel(string SLITLOTNO, bool PrintSel, string RecUpd)
        {
            using (var _Dal = new DAL.LotSlitting())
            {
                string str = System.Web.HttpContext.Current.Session["gstrUserID"].ToString();
                string result = _Dal.UpdPrintSel(SLITLOTNO, PrintSel, RecUpd, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString());

                if (result == "1")
                {
                    _Dal.Commit();
                }
                else
                {
                    _Dal.Rollback();
                }
                return result;
            }
        }

        public static string UpdPrintSelAll(bool PrintSel, string RecUpd, string filter, string filterfield, string addCondition, string passType)
        {
            using (var _Dal = new DAL.LotSlitting())
            {
                string str = System.Web.HttpContext.Current.Session["gstrUserID"].ToString();
                string result = _Dal.UpdPrintSelAll(PrintSel, RecUpd, filter, filterfield, addCondition, passType, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString());

                if (result == "1")
                {
                    _Dal.Commit();
                }
                else
                {
                    _Dal.Rollback();
                }
                return result;
            }
        }
    }
}
